### PR TITLE
Move Reflection.Emit Tokens to shared partition

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems [hunk 1]
```diff
@@ -423,15 +423,23 @@
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\CustomAttributeFormatException.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\DefaultMemberAttribute.cs" />
     <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\AssemblyBuilderAccess.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\EventToken.cs" />
+    <Compile Include="$(MSBuildThisFileDirectory)System\Reflection\Emit\FieldToken.cs" />
```
Comment (filipnavara) [inline src/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: As far as I can tell these don't exist in CoreRT now. Would it make sense to guard this with some new condition (eg. `FeatureReflectionEmit`)?

Comment (MichalStrehovsky) [reply src/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: If we added it to .NET Standard, CoreRT will be happy to pick this up. I don't see it here, but maybe it's not up to date: https://apisof.net/catalog/System.Reflection.Emit.FieldToken

Comment (filipnavara) [reply src/System.Private.CoreLib/shared/System.Private.CoreLib.Shared.projitems]: Looks like it is in .NET Standard 2.1 - https://github.com/dotnet/standard/blob/master/src/netstandard/ref/System.Reflection.Emit.cs

File: src/System.Private.CoreLib/shared/System/Reflection/Emit/StringToken.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        // Returns the metadata token for this particular string.  
+        // Generated by a call to Module.GetStringConstant().
+        //
+        public int Token { get; }
+
+        public override int GetHashCode() => Token;
+
+        public override bool Equals(object obj) => obj is StringToken st && Equals(st);
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Reflection/Emit/StringToken.cs]: The IL and resulting asm for `obj is SomeValueType svt && Equals(svt)` has been worse than `obj is SomeValueType && Equal((SomeValueType)obj)`.  I'm told that it's improved in the C# 8 compiler, but I've not verified.  I don't know whether these StringTokens show up on fast paths, but if they do, it'd be good to validate that this has no negative impact on code quality.

Comment (marek-safar) [reply src/System.Private.CoreLib/shared/System/Reflection/Emit/StringToken.cs]: @stephentoub I don't think object like Equal overload is on the hot path, secondly, all other *Token.cs implementations have the same pattern.

**Pull request merged**